internal/service/network/driver/bridge.go

```diff
@@ -0,0 +1,303 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/containerd/nerdctl/pkg/lockutil"
+	"github.com/containerd/nerdctl/pkg/netutil"
+	"github.com/runfinch/finch-daemon/api/types"
+	"github.com/runfinch/finch-daemon/internal/backend"
+	"github.com/runfinch/finch-daemon/pkg/flog"
+)
+
+const (
+	FinchICCLabelIPv4           = "finch.network.bridge.enable_icc.ipv4"
+	FinchICCLabelIPv6           = "finch.network.bridge.enable_icc.ipv6"
+	BridgeICCOption             = "com.docker.network.bridge.enable_icc"
+	BridgeHostBindingIpv4Option = "com.docker.network.bridge.host_binding_ipv4"
+	BridgeNameOption            = "com.docker.network.bridge.name"
+)
+
+type bridgeDriver struct {
+	bridgeName string
+	disableICC bool
+	netClient  backend.NerdctlNetworkSvc
+	logger     flog.Logger
+	IPv6       bool
+}
+
+var _ DriverHandler = (*bridgeDriver)(nil)
+
+var NewBridgeDriver = func(netClient backend.NerdctlNetworkSvc, logger flog.Logger, IPv6 bool) (DriverHandler, error) {
+	return &bridgeDriver{
+		netClient: netClient,
+		logger:    logger,
+		IPv6:      IPv6,
+	}, nil
+}
+
+// handleBridgeDriverOptions filters unsupported options for the bridge driver.
+func (bd *bridgeDriver) HandleCreateOptions(request types.NetworkCreateRequest, options netutil.CreateOptions) (netutil.CreateOptions, error) {
+	// enable_icc, host_binding_ipv4, and bridge name network options are not supported by nerdctl.
+	// So we process these options here and filter them out from the network create request to nerdctl.
+	processUnsupportedOptions := func(original map[string]string) map[string]string {
+		opts := map[string]string{}
+		for k, v := range original {
+			switch k {
+			case BridgeHostBindingIpv4Option:
+				if v != "0.0.0.0" {
+					bd.logger.Warnf("network option com.docker.network.bridge.host_binding_ipv4 is set to %s, but it must be 0.0.0.0", v)
```

Comment:
from the resolved comment, it seems this should be error?

Reply:
I will send a follow up PR to change this behavior since this is unrelated to the current feature.